flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManagerDriver.java

```diff
@@ -158,7 +165,9 @@ public void deregisterApplication(ApplicationStatus finalStatus, @Nullable Strin
 		// In case of pod creation failures, we should wait for an interval before trying to create new pods.
 		// Otherwise, ActiveResourceManager will always re-requesting the worker, which keeps the main thread busy.
 		final CompletableFuture<Void> createPodFuture =
-				podCreationCoolDown.thenCompose((ignore) -> kubeClient.createTaskManagerPod(taskManagerPod));
+				podCreationCoolDown.thenCompose((ignore) -> kubeClient
+						.map(client -> client.createTaskManagerPod(taskManagerPod))
+						.orElse(CompletableFuture.completedFuture(null)));
```

Comment:
Since we are expecting the client to be non-empty, I think we should fail explicitly for the empty case. 
Same for the other occurrences.